### PR TITLE
[sw/silicon_creator] Key type support for Mask ROM keys

### PR DESCRIPTION
**Pull request by alphan**
This change adds key type support for Mask ROM keys and defines the following key types:
* Prod: Valid in all operational states (TEST_UNLOCKED*, PROD, PROD_END, DEV, RMA),
* Test: Valid in TEST_UNLOCKED* and RMA states, and
* Dev: Valid only in the DEV life cycle state.

Note that we don't read from OTP when the device is in a TEST_UNLOCKED state since OTP values may not be programmed yet.

We can remove the dev key type later (or now) if we decide not to use it. It should also be fairly easy to modify how we check the validity of a key. Please let me know if you have any comments/suggestions.

### CODE REVIEW

File: sw/device/silicon_creator/mask_rom/sigverify_keys_unittest.cc [hunk 1]
```diff
@@ -26,15 +29,141 @@ using ::testing::NotNull;
...
+class SigverifyKeys : public mask_rom_test::MaskRomTest {
  protected:
+  /**
+   * Sets expectations for getting the keys stored in the Mask ROM.
+   *
+   * @param keys An array of keys.
+   */
+  template <size_t N>
+  void ExpectKeysPtrGet(const sigverify_mask_rom_key_t (&keys)[N]) {
+    size_t num_keys = std::distance(std::begin(keys), std::end(keys));
```
Comment (mundaym) [inline sw/device/silicon_creator/mask_rom/sigverify_keys_unittest.cc:152]: Isn't `num_keys` just equal to `N`?

**Review by mundaym — APPROVED** [review]

File: sw/device/silicon_creator/mask_rom/sigverify_keys_unittest.cc [hunk 2]
```diff
@@ -26,15 +29,141 @@ using ::testing::NotNull;
...
+      std::distance(std::begin(kMockKeys), std::end(kMockKeys)));
+  std::iota(res.begin(), res.end(), 0);
+  return res;
+}();
+
+/**
+ * Life cycle states used in parameterized tests.
+ */
+
+const std::vector<lifecycle_state_t> kLcStatesTest{
```
[the remark this answers is not in the document]
Comment (alphan) [reply sw/device/silicon_creator/mask_rom/sigverify_keys_unittest.cc:108]: Done.

File: sw/device/silicon_creator/mask_rom/sigverify_keys_unittest.cc (hunk 1 above)
Comment (alphan) [reply sw/device/silicon_creator/mask_rom/sigverify_keys_unittest.cc:152]: Good catch! This wasn't a template initially and forgot to remove this. Done.

File: sw/device/silicon_creator/mask_rom/mock_sigverify_keys_ptrs.h [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+namespace mask_rom_test {
+namespace internal {
+
+/**
+ * Mock class for sigverify_keys_ptrs.h
+ */
+class MockSigverifyKeysPtrs : public GlobalMock<MockSigverifyKeysPtrs> {
+ public:
+  MOCK_METHOD(const sigverify_mask_rom_key_t *, rsa_keys_ptr_get, ());
```
Comment (moidx) [inline sw/device/silicon_creator/mask_rom/mock_sigverify_keys_ptrs.h:19]: nit: Should these functions use CamelCase naming to be C++ style compliant?

**Review by moidx — APPROVED** [review]

File: sw/device/silicon_creator/mask_rom/mock_sigverify_keys_ptrs.h (hunk 3 above)
Comment (alphan) [reply sw/device/silicon_creator/mask_rom/mock_sigverify_keys_ptrs.h]: Yes, done. Thanks!

**Pull request merged**